cmd/operator_install.go

```diff
@@ -0,0 +1,206 @@
+	// and all subcommands, e.g.:
+	// operatorCmd.PersistentFlags().String("foo", "", "A help for foo")
+
+	// Cobra supports local flags which will only run when this command
+	// is called directly, e.g.:
+	// operatorCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+	operatorInstallCmd.Flags().StringVarP(&operatorManagerConfig, "config", "c", "",
+		"The configuration applied to OceanCD resources and their dependencies.")
+
+}
+
+func runOperatorInstallCmd(ctx context.Context, cmd *cobra.Command) error {
+	pathToConfig, err := cmd.Flags().GetString("config")
+	if err != nil {
+		return fmt.Errorf("failed to parse --config flag: %w", err)
+	}
+
+	installOptions := utils.Options{
+		SingleOnly:   true,
+		PathToConfig: pathToConfig,
+	}
+
+	configHandler, err := utils.NewConfigHandler(installOptions)
+	if err != nil {
+		return fmt.Errorf("failed to initiate config handler: %w", err)
+	}
+
+	err = configHandler.Handle(ctx, installOperator)
+	if err != nil {
+		return fmt.Errorf("failed to execute config handler: %w", err)
+	}
+
+	fmt.Printf("operator installation finished succesfully.\n")
+	return nil
+}
+
+func validateOperatorInstallFlags(cmd *cobra.Command) error {
+	if cmd.Flags().Lookup("config").Changed == false {
+		return nil
+	}
+
+	pathToConfig, err := cmd.Flags().GetString("config")
+	if err != nil {
+		return fmt.Errorf("failed to parse --config flag: %w", err)
+	}
+
+	if pathToConfig == "" {
+		return fmt.Errorf("path to config file must be specified")
+	}
+
+	fileExtensionWithDot := filepath.Ext(pathToConfig)
+	if err := utils.IsFileTypeSupported(fileExtensionWithDot); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func installOperator(ctx context.Context, data map[string]interface{}) error {
+
+	config, err := operator.NewInstallationConfig(data)
+	if err != nil {
+		return fmt.Errorf("failed to initialize installation config: %w", err)
+	}
+
+	payload := operator.NewInstallationPayload(config)
+	output, err := oceancd.GetOMInstallationManifests(ctx, payload)
+	if err != nil {
+		return fmt.Errorf("failed to fetch installation resources: %w", err)
+	}
+
+	resources, err := helpers.ConvertToUnstructuredSlice(output.OM.Manifests)
+	if err != nil {
+		return fmt.Errorf("failed to convert manifests to unstructured: %w", err)
+	}
+
+	operatorManagerConfigMap, err := buildOperatorManagerConfigMap(config)
+	if err != nil {
+		return fmt.Errorf("failed to build operator manager ConfigMap: %w", err)
+	}
+
+	configMapResource, err := convertOperatorManagerConfigMap(operatorManagerConfigMap)
+	if err != nil {
+		return fmt.Errorf("failed to convert operator manager ConfigMap: %w", err)
+	}
+
+	resources = append(resources, configMapResource)
+
+	applyHandler := cluster.BaseApplyHandler{}
+	for _, resource := range resources {
+		if err := applyHandler.Apply(resource); err != nil {
+			return fmt.Errorf("failed to apply operator manager ConfigMap: %w", err)
```

Comment:
the error message has typo, you can apply other resources and not only ConfigMap